### PR TITLE
Diffview support

### PR DESCRIPTION
**Pull request by mrparalon**
Related to #49 

I'm not sure, that I've thought through all corner cases, but looks like it good enough for the first draft

### CODE REVIEW

**Review by harrisoncramer — COMMENTED** [review]
This is great, just a few small suggestions. Thank you for doing this.

Do you want me to update the README after this is merged?

File: lua/gitlab/reviewer/diffview.lua [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    bufnr = nil,
+    tabnr = nil
+}
+
+-- Public Functions
+-- These functions are exposed externally and are used
+-- when the reviewer is consumed by other code. They must follow the specification
+-- outlined in the reviewer/init.lua file
+M.open         = function()
+    print("OPENING diffview")
```
[the remark this answers is not in the document]
Comment (mrparalon) [reply lua/gitlab/reviewer/diffview.lua]: Done

File: lua/gitlab/reviewer/diffview.lua [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    vim.api.nvim_command(string.format("DiffviewOpen %s", state.INFO.target_branch))
+    M.tabnr = vim.api.nvim_get_current_tabpage()
+end
+
+M.jump         = function(file_name, new_line, old_line)
+    vim.api.nvim_set_current_tabpage(M.tabnr)
```
[the remark this answers is not in the document]
Comment (mrparalon) [reply lua/gitlab/reviewer/diffview.lua:24]: Done

File: lua/gitlab/reviewer/diffview.lua [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    local file_name = nil
+    local current_line_changes = nil
+    if layout.a.file.bufnr == bufnr then
+        file_name = layout.a.file.path
+        current_line_changes = { new_line = nil, old_line = vim.api.nvim_win_get_cursor(0)[1] }
+        return file_name, current_line_changes
+    elseif layout.b.file.bufnr == bufnr then
+        file_name = layout.b.file.path
+        current_line_changes = { new_line = vim.api.nvim_win_get_cursor(0)[1], old_line = nil }
+        vim.print(file_name, current_line_changes)
```
[the remark this answers is not in the document]
Comment (mrparalon) [reply lua/gitlab/reviewer/diffview.lua]: Done

**Comment by mrparalon** [comment]
Thanks for your review! I deleted all prints and added extra nil checks

**Comment by harrisoncramer** [comment]
LGTM! Merging now.

**Pull request merged**